extensions/geode-modules/src/distributedTest/java/org/apache/geode/modules/util/ClientServerSessionCacheDUnitTest.java

```diff
@@ -105,7 +105,7 @@ public void startingAClientWithoutServersFails() {
     final VM client = VM.getVM(2);
 
     assertThatThrownBy(() -> client.invoke(this::startClientSessionCache))
-        .hasCauseInstanceOf(FunctionException.class);
+        .hasCauseInstanceOf(NoAvailableServersException.class);
```

Comment:
Now that this changes the exception thrown... will this affect any other parts of function service?  such as retries?

Reply:
This is a test code change. I don't think it will affect the product behavior of function service. 

When there is no server in the cluster, I don't think there is retry for function service. It should throw `NoAvailableServersException`.

The NullPointerException(NPE) bug (GEODE-7746) can be reproduced by this test from develop branch without any code change. The test originally expects a `FunctionException` which wraps a NPE as its cause. The NPE is exactly GEODE-7746, same stack trace. The purpose of the test to test the case of function execution when there is no server. It should throw `NoAvailableServersException`(`ServerConnectivityException`), instead of `FunctionException` which wraps a NPE. The related catch and throw code is https://github.com/apache/geode/blob/develop/geode-core/src/main/java/org/apache/geode/internal/cache/execute/ServerFunctionExecutor.java#L182-L190
and https://github.com/apache/geode/blob/develop/geode-core/src/main/java/org/apache/geode/internal/cache/execute/ServerFunctionExecutor.java#L227-L235